src/esp/bindings/SensorBindings.cpp

```diff
@@ -248,7 +251,102 @@ void initSensorBindings(py::module& m) {
+                                 allocate!) */
+                      sizeof(uint32_t), /* Size of one item */
+                      py::format_descriptor<uint32_t>::value, /* Buffer format
+                                                               */
+                      2, /* How many dimensions? */
+                      {self.specification()->resolution[0],
+                       self.specification()
+                           ->resolution[1]}, /* Number of elements for
+                                               each dimension */
+                      {sizeof(uint32_t) * self.specification()->resolution[1],
+                       sizeof(uint32_t)} /* Strides for each dimension */
+                      ));
+                  py::setattr(handle, "__obs_buffer", pyBuffer);
+                } else if (self.specification()->sensorType ==
+                           esp::sensor::SensorType::Depth) {
+                  auto pyBuffer = py::array(py::buffer_info(
+                      nullptr,       /* Pointer to data (nullptr -> ask NumPy to
+                                        allocate!) */
+                      sizeof(float), /* Size of one item */
+                      py::format_descriptor<float>::value, /* Buffer format */
+                      2, /* How many dimensions? */
+                      {self.specification()->resolution[0],
+                       self.specification()
+                           ->resolution[1]}, /* Number of elements for
+                                               each dimension */
+                      {sizeof(float) * self.specification()->resolution[1],
+                       sizeof(float)} /* Strides for each dimension */
+                      ));
+                  py::setattr(handle, "__obs_buffer", pyBuffer);
+                } else {
+                  auto pyBuffer = py::array(py::buffer_info(
+                      nullptr, /* Pointer to data (nullptr -> ask NumPy to
+                                  allocate!) */
+                      sizeof(uint8_t), /* Size of one item */
+                      py::format_descriptor<uint8_t>::value, /* Buffer format */
+                      3, /* How many dimensions? */
+                      {self.specification()->resolution[0],
+                       self.specification()->resolution[1],
+                       self.specification()->channels}, /* Number of elements
+                                                           for each dimension */
+                      {sizeof(uint8_t) * self.specification()->resolution[1] *
+                           self.specification()->channels,
+                       sizeof(uint8_t) * self.specification()->channels,
+                       sizeof(uint8_t)} /* Strides for each dimension */
+                      ));
+                  py::setattr(handle, "__obs_buffer", pyBuffer);
+                }
+              }
+            }
+            return py::getattr(handle, "__obs_buffer");
```

Comment:
This binding does not look to be covered by any current python test in sim. I need to address that.